connectors/sources/sharepoint_online.py

```diff
@@ -1148,9 +1289,10 @@ async def get_docs(self, filtering=None):
                 site_collection["siteCollection"]["hostname"],
                 self.configuration["site_collections"],
             ):
-                yield site, None
+                access_control = await self._site_access_control(site)
+                yield self._decorate_with_access_control(site, access_control), None
 
-                async for site_drive in self.site_drives(site):
+                async for site_drive in self.site_drives(site, access_control):
```

Comment:
I don't think it's necessary to pass `access_control` to `site_drives`, as well as `site_lists` and `site_pages`. It can be decorated in `get_docs` and `get_docs_incrementally`.